sale_coupon_limit/models/sale_coupon_program.py

```diff
@@ -15,11 +15,13 @@ def _check_promo_code(self, order, coupon_code):
         # The module sale_couopon_selection_wizard works with new records to probe
         # if a promotion is applicable before apply it for sure. Thus we need to ensure
         # the right id in the domain.
-        domain = [("id", "!=", order._origin.id), ("state", "!=", "cancel")] + (
-            [("promo_code", "=", coupon_code)]
-            if coupon_code
-            else [("no_code_promo_program_ids", "in", self.ids)]
-        )
+        domain = [
+            ("id", "!=", order._origin.id),
+            ("state", "!=", "cancel"),
+            "|",
+            ("no_code_promo_program_ids", "in", self.ids),
+            ("code_promo_program_id", "=", self.id),
```

Comment:
no `coupon_code` check now?

Reply:
@chienandalu can you answer this?

Reply:
Well, `promo_code` is just a relation to the correspondant `code_promo_program_id` record.